assembl/views/traversal.py

```diff
@@ -243,10 +243,13 @@ def create_query(self, id_only=True, tombstones=False):
         return query
 
     def get_class(self, typename=None):
+        """Returns the collection class, or subclass designated by typename"""
+        cls = self._class
         if typename is not None:
-            return get_named_class(typename)
-        else:
-            return self.collection.collection_class
+            other_cls = get_named_class(typename)
+            if other_cls and issubclass(other_cls, cls):
```

Comment:
I'm in /data/SomeClass, so there's a class constraint, but my request also constrains the typename further. If the typename is not a subclass of SomeClass, I should get an empty result. (contradictory class constraints)